### PR TITLE
[api] Move `Univ` out of API.

### PR DESCRIPTION
**Pull request by ejgallego**
We need to determine what part of the API should do in internal.
I'm afraid I am not qualified for that, so comments welcome.

c.f. issue #6008

### CODE REVIEW

**Comment by ejgallego** [comment]
@SkySkimmer I think here is a better example on what to put in `Internal`.

File: kernel/univ.mli [hunk 1]
```diff
@@ -406,96 +415,105 @@ sig
...
 
 (* deprecated: use qualified names instead *)
-val compare_levels : universe_level -> universe_level -> int
-val eq_levels : universe_level -> universe_level -> bool
+val compare_levels : Level.t -> Level.t -> int
+val eq_levels : Level.t -> Level.t -> bool
 
 (** deprecated: Equality of formal universe expressions. *)
-val equal_universes : universe -> universe -> bool
+val equal_universes : Universe.t -> Universe.t -> bool
```
Comment (SkySkimmer) [inline kernel/univ.mli]: Should these be ocaml.deprecated?

Comment (ejgallego) [reply kernel/univ.mli]: Good catch!

**Comment by Zimmi48** [comment]
> I think here is a better example on what to put in Internal.

WDYM? You also put `(* Nothing is internal for now *)` here.

**Comment by ejgallego** [comment]
> WDYM? You also put (* Nothing is internal for now *) here.

I mean that maybe some parts of these module shouldn't be exposed to plugins? But I am not sure which ones.

**Comment by ejgallego** [comment]
I'm closing this a for now we will only add `Internal` to certain modules and it turns out that `Univ` has no private components.

**Pull request closed without merging**